File: app/src/main/java/com/example/appskimia/Activity/MainActivity.java
package com.example.appskimia.Activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.app.Dialog;
import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.ImageButton;
import android.widget.TextView;

import com.example.appskimia.Adapter.AdapterSnapGeneric;
import com.example.appskimia.BuildConfig;
import com.example.appskimia.Data.DataGenerator;
import com.example.appskimia.Helper.StartSnapHelper;
import com.example.appskimia.Model.Image;
import com.example.appskimia.R;
import com.example.appskimia.Util.SharedPrefManager;
import com.example.appskimia.Util.Tools;
import com.freshchat.consumer.sdk.Freshchat;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    private CardView cvPetunjuk, cvKompetensi, cvMateri, cvEvaluasi;
    SharedPrefManager sharedPrefManager;
    Context mContext;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initToolbar();
        initComponent();
        initAction();
    }

    private void initAction() {
        cvPetunjuk.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showDialogAbout();
            }
        });

        cvKompetensi.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(), KompetensiActivity.class));
            }
        });

        cvEvaluasi.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(), EvaluasiActivity.class));
            }
        });

        cvMateri.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(), MateriActivity.class));
            }
        });

    }

    private void initComponent() {

        mContext = this;

        cvPetunjuk = findViewById(R.id.cv_petunjuk);
        cvKompetensi = findViewById(R.id.cv_kompetensi);
        cvMateri = findViewById(R.id.cv_materi);
        cvEvaluasi = findViewById(R.id.cv_evaluasi);

        RecyclerView recyclerStart = findViewById(R.id.recyclerStart);
        recyclerStart.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false));
        LinearLayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false);
        layoutManager.setReverseLayout(true);
        recyclerStart.setLayoutManager(layoutManager);

        // generate data
        List<Image> items = DataGenerator.getImageDate(this);
        //set data and list adapter
        recyclerStart.setAdapter(new AdapterSnapGeneric(this, items, R.layout.item_snap_basic));
        recyclerStart.setOnFlingListener(null);
        new StartSnapHelper().attachToRecyclerView(recyclerStart);
    }

    private void showDialogAbout() {
        final Dialog dialog = new Dialog(this);
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE); // before
        dialog.setContentView(R.layout.dialog_about);
        dialog.setCancelable(true);
        dialog.getWindow().setBackgroundDrawable(new ColorDrawable(android.graphics.Color.TRANSPARENT));

        WindowManager.LayoutParams lp = new WindowManager.LayoutParams();
        lp.copyFrom(dialog.getWindow().getAttributes());
        lp.width = WindowManager.LayoutParams.WRAP_CONTENT;
        lp.height = WindowManager.LayoutParams.WRAP_CONTENT;

        ((TextView) dialog.findViewById(R.id.tv_version)).setText("Version " + BuildConfig.VERSION_NAME);

        ((View) dialog.findViewById(R.id.bt_getcode)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                /*
                Intent i = new Intent(Intent.ACTION_VIEW);
                i.setData(Uri.parse("https://codecanyon.net/user/dream_space/portfolio"));
                startActivity(i);
                */
            }
        });

        ((ImageButton) dialog.findViewById(R.id.bt_close)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dialog.dismiss();
            }
        });


        dialog.show();
        dialog.getWindow().setAttributes(lp);
    }

    private void initToolbar() {
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        toolbar.setTitle("Chemistry Apps");
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        Tools.setSystemBarColor(this, R.color.grey_5);
        Tools.setSystemBarLight(this);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_search_setting, menu);
        Tools.changeMenuIconColor(menu, getResources().getColor(R.color.grey_60));
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        } else {
            Freshchat.resetUser(getApplicationContext());
            sharedPrefManager.saveSPBoolean(SharedPrefManager.SP_SUDAH_LOGIN, false);
            //startActivity(new Intent(MainActivity.this, LoginActivity.class)
            //        .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK));
            finish();
            //Toast.makeText(getApplicationContext(), item.getTitle(), Toast.LENGTH_SHORT).show();
        }
        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/com/example/appskimia/Activity/MainActivity.kt
package com.example.appskimia.Activity

import android.app.Dialog
import android.content.Context
import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.View
import android.view.Window
import android.view.WindowManager
import android.widget.ImageButton
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.appskimia.Adapter.AdapterSnapGeneric
import com.example.appskimia.BuildConfig
import com.example.appskimia.Data.DataGenerator
import com.example.appskimia.Helper.StartSnapHelper
import com.example.appskimia.R
import com.example.appskimia.Util.SharedPrefManager
import com.example.appskimia.Util.Tools
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private var cvPetunjuk: CardView? = null
    private var cvKompetensi: CardView? = null
    private var cvMateri: CardView? = null
    private var cvEvaluasi: CardView? = null
    var sharedPrefManager: SharedPrefManager? = null
    var mContext: Context? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initToolbar()
        initComponent()
        initAction()
    }

    private fun initAction() {
//        cvPetunjuk!!.setOnClickListener {
//            //showDialogAbout();
//            startActivity(Intent(applicationContext, KasusActivity::class.java))
//        }
        cvKompetensi!!.setOnClickListener { startActivity(Intent(applicationContext, KompetensiActivity::class.java)) }
        cvEvaluasi!!.setOnClickListener { startActivity(Intent(applicationContext, EvaluasiActivity::class.java)) }
        cvMateri!!.setOnClickListener { startActivity(Intent(applicationContext, MateriActivity::class.java)) }
        cv_profil.setOnClickListener { startActivity(Intent(applicationContext, ProfileActivity::class.java)) }
        cv_daftar_pustaka.setOnClickListener { startActivity(Intent(applicationContext, DaftarPustakaActivity::class.java)) }
    }

    private fun initComponent() {
        mContext = this
        //cvPetunjuk = findViewById(R.id.cv_petunjuk)
        cvKompetensi = findViewById(R.id.cv_kompetensi)
        cvMateri = findViewById(R.id.cv_materi)
        cvEvaluasi = findViewById(R.id.cv_evaluasi)
        val recyclerStart = findViewById<RecyclerView>(R.id.recyclerStart)
        recyclerStart.layoutManager = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
        val layoutManager = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
        layoutManager.reverseLayout = true
        recyclerStart.layoutManager = layoutManager
        // generate data
        val items = DataGenerator.getImageDate(this)
        //set data and list adapter
        recyclerStart.adapter = AdapterSnapGeneric(this, items, R.layout.item_snap_basic)
        recyclerStart.onFlingListener = null
        StartSnapHelper().attachToRecyclerView(recyclerStart)
    }

    private fun showDialogAbout() {
        val dialog = Dialog(this)
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE) // before
        dialog.setContentView(R.layout.dialog_about)
        dialog.setCancelable(true)
        dialog.window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        val lp = WindowManager.LayoutParams()
        lp.copyFrom(dialog.window!!.attributes)
        lp.width = WindowManager.LayoutParams.WRAP_CONTENT
        lp.height = WindowManager.LayoutParams.WRAP_CONTENT
        (dialog.findViewById<View>(R.id.tv_version) as TextView).text = "Version " + BuildConfig.VERSION_NAME
        (dialog.findViewById(R.id.bt_getcode) as View).setOnClickListener {
            /*
                Intent i = new Intent(Intent.ACTION_VIEW);
                i.setData(Uri.parse("https://codecanyon.net/user/dream_space/portfolio"));
                startActivity(i);
                */
        }
        (dialog.findViewById<View>(R.id.bt_close) as ImageButton).setOnClickListener { dialog.dismiss() }
        dialog.show()
        dialog.window!!.attributes = lp
    }

    private fun initToolbar() {
        val toolbar = findViewById<View>(R.id.toolbar) as Toolbar
        toolbar.title = "Pembelajaran Kimia"
        setSupportActionBar(toolbar)
        getSupportActionBar()?.setDisplayHomeAsUpEnabled(true)
        Tools.setSystemBarColor(this, R.color.grey_100)
        Tools.setSystemBarLight(this)
    }
}